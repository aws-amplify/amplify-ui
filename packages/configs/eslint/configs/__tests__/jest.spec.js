const jestConfig = require('../react');

describe('jest config', () => {
  it('has the expected configuration', () => {
    expect(jestConfig).toMatchSnapshot();
  });
});
