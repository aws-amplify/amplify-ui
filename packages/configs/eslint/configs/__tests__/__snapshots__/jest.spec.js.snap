// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest config has the expected configuration 1`] = `
{
  "extends": [
    "plugin:react/recommended",
    "plugin:react-hooks/recommended",
    "@aws-amplify/amplify-ui",
  ],
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true,
    },
  },
  "plugins": [
    "react",
    "react-hooks",
  ],
  "rules": {
    "react-hooks/exhaustive-deps": "error",
    "react-hooks/rules-of-hooks": "error",
    "react/destructuring-assignment": [
      "error",
      "always",
    ],
    "react/jsx-boolean-value": "error",
    "react/jsx-no-constructed-context-values": "error",
    "react/jsx-no-useless-fragment": [
      "error",
      {
        "allowExpressions": true,
      },
    ],
    "react/jsx-props-no-spreading": "off",
    "react/jsx-wrap-multilines": [
      "error",
      {
        "declaration": "ignore",
      },
    ],
    "react/no-array-index-key": "off",
    "react/no-danger": "error",
    "react/no-unused-prop-types": "error",
    "react/prop-types": "off",
    "react/static-property-placement": [
      "error",
      "static public field",
    ],
  },
  "settings": {
    "react": {
      "version": "detect",
    },
  },
}
`;
